### ISSUE TITLE
Malformed dependencies in Gemfile.lock (1.16.3 && 1.16.4)

### ISSUE THREAD

**Issue by SampsonCrowley** [bundler/bundler#6698]
running bundle install on `sass-listen 4.0.0` with bundler `1.16.3`

results in the following lines in Gemfile.lock:

      sass-listen (4.0.0)
        rb-inotify (>= 0.9.7, >= 0.9)

the resulting lines SHOULD be:

    sass-listen (4.0.0)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)

this results in breakages when pushing to heroku due to the following error:
Downloading sass-listen-4.0.0 revealed dependencies not in the API or the lockfile (rb-fsevent (>= 0.9.4, ~> 0.9), rb-inotify (>= 0.9.7, ~> 0.9))

I tried removing all traces of bundler and re-bundling, and end up with the same result every time. Updated to `1.16.4`, and had the same issue

**Comment by colby-swandale** [comment]
Can you please share an example Gemfile and Gemfile.lock to help us look into this issue.

**Comment by SampsonCrowley** [comment]
Gemfile:
```

**Comment by SampsonCrowley** [comment]
notice the lock file says 1.16.4 but I had the same problem in both versions

**Comment by greysteil** [comment]
I'm not able to replicate this locally. I took at look at sass-listen's [gemspec](https://github.com/sass/listen/blob/master/sass-listen.gemspec) and it doesn't look like it's ever been edited to have the (incorrect) requirements listed above.

@SampsonCrowley - are you seeing the same issue on any other machines or just your own?

**Comment by prashuchaudhary** [comment]
@greysteil checked this on my local machine too. Not reproducible. Correct dependencies are installed.

**Comment by greysteil** [comment]
Interesting.

@SampsonCrowley - don’t suppose you’ve made any manual edits to the rubygems cache files on your machine? That would be the obvious way to cause this. Can you try deleting your `~/.bundle/cache` folder and see if the problem persists after that?

**Comment by SampsonCrowley** [comment]
@greysteil it definitely isn't an issue with `sass-listen`s gemspec (that was the first thing I checked).

I have never touched the bundle cache on any machine.

It seems to only be that one dependency. and only on that one computer as far as I can tell.. I manually edited the lock file shortly after opening this issue to have the correct info, and the problem hasn't shown back up since.

Maybe just an un-reproducible freak accident during a bundle install that had to be manually fixed?
